### ISSUE TITLE
Mouse-press blocks events from other elements

### ISSUE THREAD

**Issue by Nelyo** [nifty-gui/nifty-gui#321]
When I click and hold my mouse on one panel and pull to the other panel, the event from the start element is the only one that throw events. But what I like to get is the mouseover-event from the element which I pass, when its happened. Whether I'm not anymore on the pressed panel or not don't change the current mouseover-element.

The same applies to released-event.
Press Panel1 -> Released in Panel2
throw me: 
Panel1 Pressed -> Panel 1 released.

```
public class StartController implements ScreenController
{
@Override
public void bind(Nifty nifty, Screen screen)
{
    Element element = screen.findElementByName("base");

    PanelBuilder panelBuilder1 = new PanelBuilder("panel1");
    panelBuilder1.backgroundColor("#ff0000");
    panelBuilder1.x("0%");
    panelBuilder1.y("0%");
    panelBuilder1.width("50%");
    panelBuilder1.height("100%");
    panelBuilder1.interactOnMouseOver("over(panel1)");
    panelBuilder1.interactOnRelease("release(panel1)");
    panelBuilder1.interactOnClickMouseMove("move(panel1)");
    panelBuilder1.build(nifty, screen, element);

    PanelBuilder panelBuilder2 = new PanelBuilder("panel2");
    panelBuilder2.backgroundColor("#0000ff");
    panelBuilder2.x("50%");
    panelBuilder2.y("0%");
    panelBuilder2.width("50%");
    panelBuilder2.height("100%");
    panelBuilder2.interactOnMouseOver("over(panel2)");
    panelBuilder2.interactOnRelease("release(panel2)");
    panelBuilder2.interactOnClickMouseMove("move(panel2)");  
    panelBuilder2.build(nifty, screen, element);
}

public void move(String id)
{
    System.out.println("Move Id: " + id);
}

public void over(String id)
{
    System.out.println("Over Id: " + id);
}

public void release(String id)
{
    System.out.println("Release Id: " + id);
}

@Override
public void onStartScreen()
{

}

@Override
public void onEndScreen()
{

}
}
```

**Comment by void256** [comment]
Yes, currently when you have an element that captured the mouse (which all mouse events do) then all events will be forwarded to that element only.

The code that's responsible for that is here: https://github.com/void256/nifty-gui/blob/1.4/nifty-core/src/main/java/de/lessvoid/nifty/screen/Screen.java#L374

Right now, I'm running out of time to investigate that further but I'll take a look if there could be some special case added for mouse over events. On a first glance I'm not sure how to add that easily though.

**Comment by dominic-jones** [comment]
I was investigating a similar issue (while trialing Nifty). On clicking a gui!button, holding the mouse!button, dragging away, and release mouse!button, the mouse release event is still triggered on the gui!button. This is at odds with other gui frameworks where the release event should normally be dropped.

The primary logic for this appears to be in ElementInteractionClickHandler where it checks for hasMouseAccess (which is a boolean deriving from FocusHandler#hasExclusiveMouseFocus). 

Ideally for my case, I don't want that hasMouseAccess check at all. But a test to disable that puts the gui into a state where it ignores all events until it receives a ButtonReleaseEvent for the original element.

While my specific case is different, the underlying issue is the same. Between mouse click and mouse release, only the initially clicked element seems capable of receiving events. I however could not track down the code that 'enabled' the gui again after the ButtonReleaseEvent fired. It was not simply in ElementInteractionClickHandler, it seemed to be relying on the ButtonReleaseEvent itself.

(Hopefully this analysis is of help.)
